### PR TITLE
wrong url for sites fixed & gutenberg editor disabled for now.

### PR DESCRIPTION
**Pull request by smarteist**
Hi , thank you for your nice and clean code. i fixed little issue who i faced in my site ,  This was due to the incorrect image addressing, which always returns the current site address instead of the global site address.
Another problem with the plugin is not working with the gutenberg editor, which I have disabled it for now and i hope to fix it in the future.

Best Regards, Ali.

### CODE REVIEW

File: src/Attachment.php [hunk 1]
```diff
@@ -57,16 +61,14 @@ public function prepareAttachmentForJs(array $response): array
...
+     * @since   2015-01-26
      */
     public function ajaxQueryAttachments()
     {
         // phpcs:disable WordPress.CSRF.NonceVerification.NoNonceVerification
         // phpcs:disable
-        $query = isset($_REQUEST['query'])
-            ? (array)wp_unslash($_REQUEST['query'])
-            : [];
+        $query = isset($_REQUEST['query']) ? (array)$_REQUEST['query'] : array();
```
Comment (bueltge) [inline src/Attachment.php]: We write the plugin for php >=7.0.0, so please change to the new art of array string `[]`. Thanks!

File: src/Plugin.php [hunk 2]
```diff
@@ -13,47 +13,97 @@ class Plugin
...
-        if (\function_exists('wc')) {
-            $this->wcBootstrap($site, $singleSwitcher);
+
+
+        add_action('admin_enqueue_scripts', [$this->assets, 'enqueueScripts']);
+        add_action('admin_enqueue_scripts', [$this->assets, 'enqueueStyles']);
+        add_action('wp_ajax_query-attachments', [$this->attachment, 'ajaxQueryAttachments'], 0);
+        add_action('wp_ajax_get-attachment', [$this->attachment, 'ajaxGetAttachment'], 0);
+        add_action('wp_ajax_send-attachment-to-editor', [$this->attachment, 'ajaxSendAttachmentToEditor'], 0);
+        add_action('wp_get_attachment_image_src', [$this->attachment, 'attachmentImageSrc'], PHP_INT_MAX, 4);
```
Comment (bueltge) [inline src/Plugin.php]: PHP_INT_MAX is not a solid option, especially for 3th level code there hook in our way. So please use her a integer, like 99. Thanks.

File: src/Attachment.php (hunk 1 above)
Comment (smarteist) [reply src/Attachment.php]: It's done ,Unfortunately tests are not copied when the project is forked by me, this is due to the lack of test files. :+1:

File: src/Site.php [hunk 3]
```diff
@@ -47,4 +47,16 @@ public function isMediaSite(): bool
...
         return ($this->id() === (int)$GLOBALS['current_blog']->blog_id);
     }
+
+    /**
+     * Replaces current site URLs with Global Site URLs
+     * It Prevents thumbnail images from not being loaded .
+     * @param $url string current site url
+     * @return string fixed url
+     */
+    public function replaceGlobalUrlPath($url)
```
Comment (widoz) [inline src/Site.php]: The method signature is missing the param type int and return type hint

File: src/WooCommerce/Gallery.php [hunk 4]
```diff
@@ -16,13 +16,11 @@ class Gallery
...
 
-    const FILTER_ATTACHMENT_IMAGE_SRC = 'wp_get_attachment_image_src';
-
     const DEFAULT_PRODUCT_TYPE = 'simple';
     const POST_PRODUCT_TYPE = 'product-type';
-    const POST_PRODUCT_IMAGE_GALLERY = 'product_image_gallery';
-
     const META_KEY_PRODUCT_GALLERY = '_product_image_gallery';
+    const POST_PRODUCT_IMAGE_GALLERY = 'product_image_gallery';
+    const FILTER_ATTACHMENT_IMAGE_SRC = 'wp_get_attachment_image_src';
```
Comment (widoz) [inline src/WooCommerce/Gallery.php]: This seems to me additional unnecessary changes.

File: src/Thumbnail.php [hunk 5]
```diff
@@ -140,10 +140,13 @@ public function adminPostThumbnailHtml(string $content, int $postId, $attachment
             $removeImageLabel = $postTypeObject->labels->remove_featured_image;
         }
 
-        return $this->replaceRemovePostThumbnailMarkup(
+        $content = $this->replaceRemovePostThumbnailMarkup(
             $removeImageLabel,
             $content
         );
+
+        return $this->site->replaceGlobalUrlPath($content);
```
Comment (widoz) [inline src/Thumbnail.php]: I understanding this is the main fix for the PR but I haven't enough informations from your PR comment, could you help in reproduce the problem? May be by adding the steps to reproduce it?

**Comment by bueltge** [comment]
@smarteist Maybe you are not familiar with a revert. see this simple how to https://help.github.com/en/articles/reverting-a-pull-request - possible on a github page.

**Comment by bueltge** [comment]
@smarteist why you close them? It is not more relevant or to much effort for you in context of our review? Please a short feedback, maybe your pr is helpful and we should implement in different prs, commits. Thanks.
